Spigot-Server-Patches/0248-Send-attack-SoundEffects-only-to-players-who-can-see.patch

```diff
@@ -0,0 +1,102 @@
+                         ++i;
+                         flag1 = true;
+                     }
+@@ -1041,7 +1052,7 @@ public abstract class EntityHuman extends EntityLiving {
+                                 }
+                             }
+ 
+-                            this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, SoundEffects.fz, this.bK(), 1.0F, 1.0F);
++                            sendSoundEffect(SoundEffects.fz); // Paper - send while respecting visibility
+                             this.cX();
+                         }
+ 
+@@ -1071,15 +1082,15 @@ public abstract class EntityHuman extends EntityLiving {
+                         }
+ 
+                         if (flag2) {
+-                            this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, SoundEffects.fv, this.bK(), 1.0F, 1.0F);
++                            sendSoundEffect(SoundEffects.fv); // Paper - send while respecting visibility
+                             this.a(entity);
+                         }
+ 
+                         if (!flag2 && !flag3) {
+                             if (flag) {
+-                                this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, SoundEffects.fy, this.bK(), 1.0F, 1.0F);
++                                sendSoundEffect(SoundEffects.fy); // Paper - send while respecting visibility
+                             } else {
+-                                this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, SoundEffects.fA, this.bK(), 1.0F, 1.0F);
++                                sendSoundEffect(SoundEffects.fA); // Paper - send while respecting visibility
+                             }
+                         }
+ 
+@@ -1135,7 +1146,7 @@ public abstract class EntityHuman extends EntityLiving {
+ 
+                         this.applyExhaustion(world.spigotConfig.combatExhaustion); // Spigot - Change to use configurable value
+                     } else {
+-                        this.world.a((EntityHuman) null, this.locX, this.locY, this.locZ, SoundEffects.fx, this.bK(), 1.0F, 1.0F);
++                        sendSoundEffect(SoundEffects.fx); // Paper - send while respecting visibility
+                         if (flag4) {
+                             entity.extinguish();
+                         }
+diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
+index e193f86..fe2afee 100644
+--- a/src/main/java/net/minecraft/server/World.java
++++ b/src/main/java/net/minecraft/server/World.java
+@@ -1068,6 +1068,12 @@ public abstract class World implements IBlockAccess {
+ 
+     }
+ 
++    // Paper start - OBFHELPER
++    public final void sendSoundEffect(@Nullable EntityHuman fromEntity, double x, double y, double z, SoundEffect soundeffect, SoundCategory soundcategory, float volume, float pitch) {
++        this.a(fromEntity, x, y, z, soundeffect, soundcategory, volume, pitch);
```

Comment:
Same with this one, needs to be directly above the this.a() that has the matching signature (the a below this is not the same one)